### PR TITLE
fix(cursor): prefer enterprise auth and handle missing limits

### PR DESCRIPTION
**Pull request by drewwells**
This change fixes a multi-account Cursor edge case by preferring
    keychain auth when SQLite indicates a free account and the token
    subjects differ, ensuring enterprise usage is attributed to the
    correct account. It also handles percent-only usage payloads when
    planUsage.limit is missing, avoiding false errors while still
    rendering valid usage lines.

    Regression tests were added for both auth-source selection and
    missing-limit percent usage.

## Type of Change

- [x] Bug fix
- [ ] New feature
- [ ] New provider plugin
- [ ] Documentation
- [ ] Performance improvement
- [ ] Other (describe below)

## Testing

- [x] I ran `bun run build` and it succeeded
- [x] I ran `bun run test` and all tests pass
- [x] I tested the change locally with `bun tauri dev`

## Screenshots

image

## Checklist

- [x] I read [CONTRIBUTING.md](../CONTRIBUTING.md)
- [ ] My PR targets the `main` branch
- [ ] I did not introduce new dependencies without justification

---
## Summary by cubic
Fixes a multi-account auth edge case by preferring keychain tokens when SQLite looks free and JWT subjects differ, so enterprise usage is attributed to the right account. Also handles Connect responses that omit `planUsage.limit` by using percent-only data when available and falling back to request-based metrics when needed, including team-inferred accounts.

- **Bug Fixes**
  - Prefer keychain auth when SQLite shows "free" and JWT subjects differ.
  - When `planUsage.limit` is missing: render percent usage if `totalPercentUsed` exists; otherwise fall back to REST request-based usage (Enterprise/Team, including team-inferred when plan info is unavailable).
  - Add regression tests for auth selection, percent-only handling (including missing plan info), enterprise missing-limit fallback, and team-inferred fallback.

<sup>Written for commit b821f34bd3a1b09153ec7d33a51399953105c1a3. Summary will update on new commits.</sup>

### CODE REVIEW

File: plugins/cursor/plugin.js [hunk 1]
```diff
@@ -465,14 +502,24 @@
     }
 
     const needsFallbackWithoutPlanInfo = usage.enabled !== false &&
-      !usage.planUsage &&
+      (!hasPlanUsage || planUsageLimitMissing) &&
+      !hasTotalUsagePercent &&
```
[the remark this answers is not in the document]
Comment (drewwells) [reply plugins/cursor/plugin.js:506]: I'm not sure how important this is... it seems like overengineering. We will not catch all edge cases, but I don't believe this adds a new error

**Comment by davidarny** [comment]
@codex review

**Review by robinebers — CHANGES_REQUESTED** [review]
Thanks for the update — I found one remaining edge case to fix before approval.

In , when:
-  is missing,
-  exists,
- plan info is unavailable,
- and the account is team-inferred from ,

the current flow can skip request-based fallback and then throw 

Can you update the fallback logic so this path still uses request-based fallback (or otherwise avoids hard failure), and add a regression test for this exact scenario?

Please re-request review after that.

**Review by robinebers — CHANGES_REQUESTED** [review]
Thanks for the update — I found one remaining edge case to fix before approval.

In `plugins/cursor/plugin.js`, when:
- `planUsage.limit` is missing,
- `planUsage.totalPercentUsed` exists,
- plan info is unavailable,
- and the account is team-inferred from `spendLimitUsage`,

the current flow can skip request-based fallback and then throw `Total usage limit missing from API response.`

Can you update the fallback logic so this path still uses request-based fallback (or otherwise avoids hard failure), and add a regression test for this exact scenario?

Please re-request review after that.

**Comment by drewwells** [comment]
Your comment doesn’t make any sense. Is this a chat bot you are working on? 

The ChatGPT bot wants me to prefer a free token even when an account token is found. If their credentials are expired, they will refresh them. We don’t want to show a free account in that scenario.

**Comment by drewwells** [comment]
Fed the pr comments into a cursor and let it make changes. 

The cursor IDE will use those embedded sqlite credentials if the keychain ones disappear. I didn't add verifying creds, that should probably be handled by shared logic elsewhere. I went to see if the crusor ide will actually use those sqlite creds, it does in fact fallback to the embedded sqlite creds if keychain is invalid. I imagine that's temporary while they move everybody to keychain. Enjoy the unencrypted credentials until then 🙃

**Review by robinebers — APPROVED** [review]
Rebased only and ready to merge.
